cli/src/main/java/com/devonfw/tools/ide/git/GitContextImpl.java

```diff
@@ -255,17 +258,59 @@ IdeContext getContext() {
   }
 
   @Override
-  public void verifyGitInstalled() {
-
-    this.context.findBashRequired();
-    Path git = Path.of("git");
-    Path binaryGitPath = this.context.getPath().findBinary(git);
-    if (git == binaryGitPath) {
-      String message = "Git is not installed on your computer but required by IDEasy. Please download and install git:\n"
-          + "https://git-scm.com/download/";
+  public Path findGitRequired() {
+
+    Path gitPath = findGit();
+    if (gitPath == null) {
+      String message = "Git " + IdeContext.IS_NOT_INSTALLED;
+      if (SystemInfoImpl.INSTANCE.isWindows()) {
+        message += IdeContext.PLEASE_DOWNLOAD_AND_INSTALL_GIT + ":\n "
+            + IdeContext.WINDOWS_GIT_DOWNLOAD_URL;
+      }
       throw new CliException(message);
     }
-    this.context.trace("Git is installed");
+    return gitPath;
+  }
+
+  /**
+   * Finds the path to the Git executable.
+   *
+   * @return the {@link Path} to the Git executable, or {@code null} if Git is not found.
+   */
+  public Path findGit() {
+    if (this.git != null) {
+      return this.git;
+    }
+    Path bashBinary = this.context.findBashRequired();
+    Path gitPath = Path.of("git");
+    if (SystemInfoImpl.INSTANCE.isWindows()) {
+      if (Files.exists(bashBinary)) {
+        gitPath = bashBinary.getParent().resolve("git.exe");
+        if (Files.exists(gitPath)) {
+          this.context.trace("Git path was extracted from bash path at: {}", gitPath);
+          gitPath = findGit();
+          SystemPath systemPath = this.context.getPath();
+          systemPath.setPath("git", gitPath);
+        } else {
+          this.context.error("Git path: {} was extracted from bash path at: {} but it does not exist", gitPath, bashBinary);
+          return null;
+        }
+      } else {
+        this.context.error("Bash path was checked at: {} but it does not exist", bashBinary);
+        return null;
+      }
+    } else {
+      Path binaryGitPath = this.context.getPath().findBinary(gitPath);
```

Comment:
This is the how this implementation should start after `this.git` is not initialised.
Only if `git` is not found on `PATH` we can start with our "windows hack/workaround" and also remove the code adding git to the `SystemPath`.